src/Tickets/Commerce/Communication/Email.php

```diff
@@ -31,8 +32,13 @@ public function send_tickets_email( $order_id, $post_id = null ) {
 
 		// Look at each attendee and check if a ticket was sent: in each case where a ticket
 		// has not yet been sent we should a) send the ticket out by email and b) record the
-		// fact it was sent
+		// fact it was sent.
 		foreach ( $all_attendees as $single_attendee ) {
+			// If we have a post ID, only add those attendees/tickets that are for that event.
+			if ( $post_id && (int) $single_attendee['event_id'] !== $post_id ) {
```

Comment:
The `$post_id` here is not cast to `int` in the function arguments, if you use a strict comparison, cast it first:
```suggestion
			if ( $post_id && (int) $single_attendee['event_id'] !== (int)$post_id ) {
```